lib/active_merchant/billing/gateways/d_local.rb

```diff
@@ -239,8 +281,28 @@ def error_code_from(action, response)
         code&.to_s
       end
 
+      def error_code_from_get(response)
+        unless success_from_get(response)
+          if response['error']
+            if response['error']['reason'] != 'NA'
+              response['error']['reason']
+            else
+              response['error']['code']
+            end
+          elsif response['error_reason']
+            response['error_reason']
+          elsif response['error_code']
+            response['error_code']
+          elsif response['code']
+            response['code'].to_s
+          else
+            'BAD_REQUEST_ERROR'
+          end
+        end
+      end
```

Comment:
I don't think we will need to check for `error`, `error_reason`, or `error_code` in the response - https://docs.dlocal.com/api-documentation/payins-api-reference/payments#errors.
Am I missing something?

Reply:
Modified.